### PR TITLE
Commenting webpack.config.js

### PR DESCRIPTION
**Pull request by iabril**
Commenting webpack.config.js for practices 00 Intro - 00 Boilerplate ad 02 Server

I have kept the same original file structure except webpack.config.js file, which have been commented explaining each line.

### CODE REVIEW

**Comment by iabril** [comment]
I have had new changes, commenting webpack.config.js file, explaining each line, and incorporing nodemon in package.json to avoid recharge webpack when there is changes on webpack.config.js on 00 Intro - 03 Ouput practice

File: 98 Commented/00 Intro/00 Boilerplate/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
+WEBPACK is a module bundler for modern JavaScript applications. 
+When webpack processes your application, it recursively builds a dependency graph,
+that includes every module your application needs, then packages all of those modules into one or more bundles.
+
+There are a lot of statements that have been standardized in ES2015. 
+Although they are not supported in most browsers yet, webpack does support them out of the box.
+
+Webpack "transpiles" the code so that older browsers can also run it.
```
Comment (brauliodiez) [inline 98 Commented/00 Intro/00 Boilerplate/webpack.config.js]: Nice intro, but on the last two paragraphs I would rather say:

Webpack integrated well with transpilers like babel or typescript, they allow us...

Webpack itself does not convert from ES6 to ES5 is babel or typescript the ones that do so.

File: 98 Commented/00 Intro/00 Boilerplate/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+*/
+
+module.exports = {
+    //An entry point indicates which module webpack should use to begin building out its internal dependency graph
+    // So, webpack will start reading on this entry to build our app, web, etc.
```
Comment (brauliodiez) [inline 98 Commented/00 Intro/00 Boilerplate/webpack.config.js:14]: We can add something like, in this sample app "students.js" is the starting point, usually you will find a more generica root  file name like _index.js_, _app.js_....

File: 98 Commented/00 Intro/00 Boilerplate/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  entry: ['./students.js'],
+    //The output property tells webpack where to emit the bundles it creates and how to name these files
+    // in this case, we are saing to wepack that built our app in the file: "bundle.js", and this will be
+    // the file served to clients how enter in our resources (our app, our web, etc. depending on our development)
```
Comment (brauliodiez) [inline 98 Commented/00 Intro/00 Boilerplate/webpack.config.js]: Indicate here that we can chop into several bundles, not only a single file

File: 98 Commented/00 Intro/00 Boilerplate/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        //Loaders enable webpack to process more than just JavaScript files (webpack itself only understands JavaScript).
+        //webpack loaders transform all types of files into modules that can be included in your application's dependency graph.
+    rules: [
+      // The next rule add support for es6, we will ask webpack to handle all js files under the project folder 
+      // (excluding the node_modules subfolder) and transpile them from es6 to es5 (using the babel-loader).
+      {
+          //test: Identify which file or files should be transformed by a certain loader
+          // We can use regular expressions, so here we are saying to webpack that process any files with extension starting in ".js_" with babe's loader
+        test: /\.js$/,
+          //with exclude we can say to webpack where don't proceess files to build our app/web.
```
Comment (brauliodiez) [inline 98 Commented/00 Intro/00 Boilerplate/webpack.config.js:38]: Add: in this case we are asking webpack to process js files under the _node_modules_ folder

File: 98 Commented/00 Intro/00 Boilerplate/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        exclude: /node_modules/,
+          //charge babel's loader for "transpilar" from ES6 to ES5
```
Comment (brauliodiez) [inline 98 Commented/00 Intro/00 Boilerplate/webpack.config.js]: Use _babel-loader_ to transpile from ES6 to ES5. We can use a single loader or we can chain the execution of several loaders.

**Review by iabril — COMMENTED** [review]
Ok, I think that all your comments are corrects, I'll to the changes and.. do I a new PR?

**Comment by brauliodiez** [comment]
In the same pull when you push it gets automatically updated.

On each comment go answering 'done' or asking questions or commenting, or indicating why not

Whenever you are donde give me a buzz

File: 98 Commented/00 Intro/00 Boilerplate/webpack.config.js (hunk 1 above)
Comment (iabril) [reply 98 Commented/00 Intro/00 Boilerplate/webpack.config.js]: Ok, changes written and commit done with #comment1 (this changes has been made on the three projects that I had downloaded)

File: 98 Commented/00 Intro/00 Boilerplate/webpack.config.js (hunk 2 above)
Comment (iabril) [reply 98 Commented/00 Intro/00 Boilerplate/webpack.config.js:14]: Ok, changes written and commit done with #comment2 (this changes has been made on the three projects that I had downloaded)

File: 98 Commented/00 Intro/00 Boilerplate/webpack.config.js (hunk 3 above)
Comment (iabril) [reply 98 Commented/00 Intro/00 Boilerplate/webpack.config.js]: Ok, changes written and commit done with #comment3 (this changes has been made on the three projects that I had downloaded)

File: 98 Commented/00 Intro/00 Boilerplate/webpack.config.js (hunk 4 above)
Comment (iabril) [reply 98 Commented/00 Intro/00 Boilerplate/webpack.config.js:38]: Ok, changes written and commit done with #comment4 (this changes has been made on the three projects that I had downloaded)

File: 98 Commented/00 Intro/00 Boilerplate/webpack.config.js (hunk 5 above)
Comment (iabril) [reply 98 Commented/00 Intro/00 Boilerplate/webpack.config.js]: Ok, changes written and commit done with #comment5 (this changes has been made on the three projects that I had downloaded)

**Pull request merged**